### ISSUE TITLE
tss2 tools busybox style

### ISSUE THREAD

**Issue by williamcroberts** [tpm2-software/tpm2-tools#2105]
Make these tools busybox style as well.

**Comment by AndreasFuchsSIT** [comment]
@osresearch Could you help out on this issue, since I have basically no clue what voodoo you used for this... ;-)

**Comment by williamcroberts** [comment]
He made macro that uses the constructor extension:
```
The constructor attribute causes the function to be called automatically before execution enters main (). Similarly, the destructor attribute causes the function to be called automatically after main () has completed or exit () has been called. Functions with these attributes are useful for initializing data that will be used implicitly during the execution of the program.
```

The constructor is in each tool C file, and registers a set of callbacks for the on_<xxx> framework functions that main needs to know about.

The main() function checks argv[0], and looks up the tool and:
1. chops argv[0] if the tool is called as a commandlet like tpm2 getrandom
   - I think after the left shift of argv, we need to rewrite argv[0] to tpm2_getrandom to make man 
     work  again.
2. If its argv[0] tpm2_getrandom, just use getrandom
3. Now both paths end up with key getrandom, and perform a look up for the various function pointers
4. Call those.

**Comment by osresearch** [comment]
The previous plan was even more magic.  The constructor segment approach is at least an 80% reduction in ELF magic compared to the `objcopy` symbolg renaming technique.

It should be easy to copy the code into [`tools/fapi/tss2_template.c`](https://github.com/tpm2-software/tpm2-tools/blob/master/tools/fapi/tss2_template.c#L357) to have the same sort of logic.  I haven't looked at every tool in there to verify that it uses the same callbacks, although the few that I spot checked seemed to do so.

**Comment by osresearch** [comment]
Preliminary patch: https://github.com/osresearch/tpm2-tools/tree/tss-bundle

**Comment by osresearch** [comment]
https://github.com/tpm2-software/tpm2-tools/pull/2123 is passing the tests now

**Comment by idesai** [comment]
fixed via #2123
